### ISSUE TITLE
Fail to build per channel smoothquant model for bloom model

### ISSUE THREAD

**Issue by MIrandaStock** [NVIDIA/TensorRT-LLM#496]
Using the follow commandline to build an engine for bloom model with per channel smoothquant.
```
python3 build.py --bin_model_dir $bin_model_dir \
    --use_gpt_attention_plugin float16 \
    --output_dir $engine_dir \
    --use_smooth_quant \
    --per_channel \
    --gather_all_token_logits
```
The quantization result looks very abnormal：
![image](https://github.com/NVIDIA/TensorRT-LLM/assets/45962405/28956c60-49ff-48e1-a367-61ca5e59323e)

however, other combination of --per_channel and --per_token have a good result, such as 
```
python3 build.py --bin_model_dir $bin_model_dir \
    --use_gpt_attention_plugin float16 \
    --output_dir $engine_dir \
    --use_smooth_quant \
    --gather_all_token_logits
```
or
```
python3 build.py --bin_model_dir $bin_model_dir \
    --use_gpt_attention_plugin float16 \
    --output_dir $engine_dir \
    --use_smooth_quant \
    --per_channel \
    --per_token \
    --gather_all_token_logits
```
the result looks good:
![image](https://github.com/NVIDIA/TensorRT-LLM/assets/45962405/d68575e2-9a37-4665-a84b-0f47ff7b951a)

**Comment by wm2012011492** [comment]
We only fully test the two modes: ` _per_tensor_` and `_per_token_ + _per_channel_` mode. Can they satisfy your requirements? @byshiue for viz

**Comment by byshiue** [comment]
It looks the accuracy of only per_channel is not good, but not a bug.

**Comment by MIrandaStock** [comment]
> It looks the accuracy of only per_channel is not good, but not a bug.

Got it,  thanks.

**Comment by MIrandaStock** [comment]
> It looks the accuracy of only per_channel is not good, but not a bug.

**Comment by MIrandaStock** [comment]
In theory，`per_channel` is better than `_per_tensor_`, right? So, why the following commandline can have a better quantization result? But only per_channel cannot.
```
python3 build.py --bin_model_dir $bin_model_dir \
    --use_gpt_attention_plugin float16 \
    --output_dir $engine_dir \
    --use_smooth_quant \
    --gather_all_token_logits
```
